### PR TITLE
Add file size restriction for legacy API

### PR DESCRIPTION
**Pull request by Ceredron**
## Description
Add file size restriction for legacy API. Same as A2 file size restriction.

## Related Issue(s)
- #550
## Verification
- [X] **Your** code builds clean without any errors or warnings
- [X] Manual testing done (required)
- [X] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [X] All tests run green

## Documentation
- [X] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

## Summary by CodeRabbit

- **New Features**
	- Added a `ContentLength` property to enhance file upload information.
	- Introduced a method to determine maximum upload size based on resource configuration.

- **Bug Fixes**
	- Improved file size validation logic to streamline processing.

- **Tests**
	- Added a test for validating file upload size limits, ensuring proper error handling for oversized files.
	- Introduced a `FakeFileStream` class for simulating file streams in tests.

### CODE REVIEW

File: src/Altinn.Broker.API/Controllers/LegacyFileController.cs [hunk 1]
```diff
@@ -77,7 +77,8 @@ CancellationToken cancellationToken
             FileTransferId = fileTransferId,
             Token = legacyToken,
             UploadStream = Request.Body,
-            IsLegacy = true
+            IsLegacy = true,
+            ContentLength = Request.ContentLength ?? Request.Body.Length
```
[the remark this answers is not in the document]
Comment (Andreass2) [reply src/Altinn.Broker.API/Controllers/LegacyFileController.cs:81]: Should we consider this or some other error handling? 
May be unneccessary if we know all filesizes is smaller

**Review by Andreass2 — APPROVED** [review]
I would look into some of the codeRabbit comments

File: src/Altinn.Broker.API/Controllers/LegacyFileController.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ceredron) [reply src/Altinn.Broker.API/Controllers/LegacyFileController.cs:81]: All the other comments were a bit stupid, but this one is very valid. I'm inclined to close this Pull Request as this is the Legacy API calling us (and file size is restricted there). Main purpose of this PR was to have it explicit in our code too, but not really worth it if it causes memory issues itself.

[the remark this answers is not in the document]
Comment (Ceredron) [reply src/Altinn.Broker.API/Controllers/LegacyFileController.cs:81]: On the other side, we should require content-length header to be set for the normal upload operation.

**Comment by Ceredron** [comment]
Abandoned for now

**Pull request closed without merging**